src/webapi/airportactionscontroller.cpp

```diff
@@ -0,0 +1,77 @@
+/*****************************************************************************
+* Copyright 2015-2020 Alexander Barthel alex@littlenavmap.org
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*****************************************************************************/
+
+#include "airportactionscontroller.h"
+#include "abstractlnmactionscontroller.h"
+
+#include <QDebug>
+
+AirportActionsController::AirportActionsController(QObject *parent, bool verboseParam, AbstractInfoBuilder* infoBuilder) :
+    AbstractLnmActionsController(parent, verboseParam, infoBuilder)
+{
+    qDebug() << Q_FUNC_INFO;
+}
+
+WebApiResponse AirportActionsController::infoAction(WebApiRequest request){
+    qDebug() << Q_FUNC_INFO << request.parameters.value("ident");
+
+    // Get a new response object
+    WebApiResponse response = getResponse();
+
+    // Query item
+    map::MapAirport airport = getAirportByIdent(request.parameters.value("ident").toUpper());
+
+    if(airport.id > 0){
+
+        // Fetch related data
+        const SqlRecord* airportInformation = getAirportInformation(airport.id);
+        const AirportAdminNames airportAdminNames = getAirportAdminNames(airport);
+        const int transitionAltitude = getTransitionAltitude(airport);
+
+        const QTime sunrise = getSunrise(*airportInformation);
+        const QTime sunset =  getSunset(*airportInformation);
+        const QDateTime activeDateTime = getActiveDateTime();
+        const QString activeDateTimeSource = getActiveDateTimeSource();
+
+        // Compose data container
+        AirportInfoData data = {
+            airport,
+            getWeatherContext(airport),
+            nullptr,
+            airportInformation,
+            &airportAdminNames,
+            &transitionAltitude,
+            &sunrise,
+            &sunset,
+            &activeDateTime,
+            &activeDateTimeSource
```

Comment:
Sry but for my understanding: If this method would return `data` directly its `sunrise` pointer would become invalid, right? Because `sunrise` since its declared inside, is only alive as long as this method is being executed.

Reply:
yes; C++ now allows static function variables (https://www.geeksforgeeks.org/static-keyword-cpp/). Their initialisation is only performed for the first time the function is run. This also might help with server load regarding data longevity :)
Other than this you could declare the QTime a `QTime` rather than a `QTime*` in `AirportInfoData`. Or you create a `new` QTime object here (to a `QTime*`) but then you have to keep track of it and `delete` it before its last pointer gets overwritten/lost and before program exit.

Reply:
A side note: You should enable the warnings in Qt creator. Enable the ClangCodeModel plugin and configure it in options. I use `-Wall -Wextra -Wno-cppcoreguidelines-pro-bounds-array-to-pointer-decay` for Clang warnings. These help a lot.

I've seen a warning about a method declared as returning a type and not returning it for all branches (`AbstractLnmActionsController::getAirportQuery`). Running into this means instant crash. 

Stupid Clang shows only a warning. I'd really prefer a compiler error. :frowning_face: 

```
/home/alex/Projekte/littlenavmap/src/webapi/abstractlnmactionscontroller.cpp:61: warning: control reaches end of non-void function [-Wreturn-type]
../littlenavmap/src/webapi/abstractlnmactionscontroller.cpp: In member function ‘AirportQuery* AbstractLnmActionsController::getAirportQuery(AbstractLnmActionsController::AirportQueryType)’:
../littlenavmap/src/webapi/abstractlnmactionscontroller.cpp:61:1: warning: control reaches end of non-void function [-Wreturn-type]
   61 | }
      | ^
```

Reply:
Ah, cool! Thx for the hints! I'll dive into this.